### PR TITLE
BASH-6 Automatically run Build prior to packaging

### PR DESCRIPTION
**Pull request by csduarte**
**Description**
The code may vary from the current code when packaging (generally in the case of non-automated builds), this ensures the current code is built right before packaging.

- [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

**Test Cases**
`npm run package:mac` will run build prior to package scripts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,11 +25,11 @@
     "watch:renderer": "webpack-dev-server --config webpack.config.renderer.js",
     "serve": "gulp watch",
     "test": "npm-run-all build test:* lint:*",
```
Comment (yuya-oc) [inline package.json]: Remove `build` target to avoid executing `npm run build` twice.

File: package.json [hunk 2]
```diff
@@ -25,11 +25,11 @@
...
-    "test:app": "mocha --reporter mocha-circleci-reporter --recursive test/specs",
+    "test:app": "npm run build && mocha --reporter mocha-circleci-reporter --recursive test/specs",
     "package:all": "npm-run-all package:windows package:mac package:linux",
-    "package:windows": "build --win --x64 --ia32 --em.name=mattermost --publish=never && npm run manipulate-windows-zip",
-    "package:mac": "build --mac --publish=never",
-    "package:linux": "build --linux --x64 --ia32 --em.name=mattermost-desktop --publish=never",
+    "package:windows": "npm run build && build --win --x64 --ia32 --em.name=mattermost --publish=never && npm run manipulate-windows-zip",
+    "package:mac": "npm run build && build --mac --x64 --ia32 --publish=never",
```
Comment (yuya-oc) [inline package.json]: `--x64 --ia32` is not necessary. Only x64 is always built for mac.

**Comment by yuya-oc** [comment]
In past version, I had removed `npm run build` because it runs redundantly. However, if it's not comfortable for contributors, I'm agree to reverting.

**Comment by dmeza** [comment]
@yuya-oc rebased to latest. Fixed your comments, built, packaged and run linting. Let me know if that's the correct change.

**Pull request merged**